### PR TITLE
[Docs] Add Security Best Practices to Ray 

### PR DESCRIPTION
**Pull request by ijrsvt**
## Why are these changes needed?
* Make sure our Security Best Practices are on the Top Level Navigation

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: doc/source/ray-security/index.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+When considering the security responsibilities of running Ray, it helps to think of the different entities interacting with Ray.
+* **Developers** write code that relies on Ray. They either run a single-node Ray Cluster locally or multi-node Clusters remotely on provided compute infrastructure.
+* **Platform providers** provide the compute environment on which **Developers** run Ray.
+* **Users** interact with the output of Ray-powered applications.
+
+## Best Practices
+**Security and isolation must be enforced outside of the Ray Cluster.** Ray expects to run in a safe network environment and to act upon trusted code. Developers and platform providers must maintain these invariants to ensure the safe operation of Ray Clusters:
+
+### Deploy Ray Clusters in a governed network environment
```
Comment (ijrsvt) [inline doc/source/ray-security/index.md:21]:
```suggestion
### Deploy Ray Clusters in a controlled network environment
```

File: doc/source/ray-security/index.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+## Personas
+
+When considering the security responsibilities of running Ray, it helps to think of the different entities interacting with Ray.
+* **Developers** write code that relies on Ray. They either run a single-node Ray Cluster locally or multi-node Clusters remotely on provided compute infrastructure.
+* **Platform providers** provide the compute environment on which **Developers** run Ray.
+* **Users** interact with the output of Ray-powered applications.
+
+## Best Practices
+**Security and isolation must be enforced outside of the Ray Cluster.** Ray expects to run in a safe network environment and to act upon trusted code. Developers and platform providers must maintain these invariants to ensure the safe operation of Ray Clusters:
```
Comment (angelinalg) [inline doc/source/ray-security/index.md:19]:
```suggestion
**Security and isolation must be enforced outside of the Ray Cluster.** Ray expects to run in a safe network environment and to act upon trusted code. Developers and platform providers must maintain the following invariants to ensure the safe operation of Ray Clusters.
```

Comment (angelinalg) [reply doc/source/ray-security/index.md:19]: Sorry if I'm flip flopping on the use of the colon at the end of this sentence. Seeing it rendered is making me realize that it is a regular introductory sentence.

File: doc/source/ray-security/index.md [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+### Deploy Ray Clusters in a governed network environment
+* Network traffic between Ray core components and Ray auxiliary components should always be in a controlled, isolated network. Access to auxiliary services should be gated with strict network controls and/or external authentication/authorization proxies.
```
Comment (angelinalg) [inline doc/source/ray-security/index.md:22]: Should `core` be capitalized?

File: doc/source/ray-security/index.md [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+* gRPC communication can be encrypted with TLS, but it is not a replacement for network isolation.
+* Platform providers are responsible for ensuring that Ray runs in sufficiently controlled network environments and that developers can access features like Ray Dashboard in a secure manner.
+### Only execute trusted code within Ray
+* Ray faithfully executes code that is passed to it – Ray doesn’t differentiate between a tuning experiment, a rootkit install, or an S3 bucket inspection.
+* Ray developers are responsible for building their applications with this understanding in mind.
+### Enforce isolation outside of Ray with multiple Ray Clusters
+* If workloads require isolation from each other, use separate, isolated Ray Clusters. Ray can schedule multiple distinct Jobs in a single Cluster, but does not attempt to enforce isolation between them. Similarly, Ray does not implement access controls for developers interacting with a given cluster.
```
Comment (angelinalg) [inline doc/source/ray-security/index.md:29]:
```suggestion
* If workloads require isolation from each other, use separate, isolated Ray Clusters. Ray can schedule multiple distinct Jobs in a single Cluster, but doesn't attempt to enforce isolation between them. Similarly, Ray doesn't implement access controls for developers interacting with a given cluster.
```

File: doc/source/ray-security/index.md [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+
+To achieve performant and distributed workloads, Ray components require intra-cluster communication. This communication includes central tenants like distributed memory and node-heartbeats, as well as auxiliary functions like metrics and logs. Ray leverages gRPC for a majority of this communication.
+
+Ray offers additional services to improve the developer experience. These services include Ray Dashboard (to allow for cluster introspection and debugging), Ray Jobs (hosted alongside the Dashboard and provides the ability to submit Ray Jobs) and Ray Client (to allow for local, interactive development with a remote cluster). These services provide complete access to the Ray Cluster and the underlying compute resources.
+
+## Personas
+
+When considering the security responsibilities of running Ray, it helps to think of the different entities interacting with Ray.
+* **Developers** write code that relies on Ray. They either run a single-node Ray Cluster locally or multi-node Clusters remotely on provided compute infrastructure.
```
Comment (angelinalg) [inline doc/source/ray-security/index.md:14]: I noticed that you are using `Ray developers` later on. Should this persona be called `Ray developers`?

**Review by angelinalg — APPROVED** [review]
Just some copy edits and nits.

File: doc/source/ray-security/index.md (hunk 5 above)
Comment (ijrsvt) [inline doc/source/ray-security/index.md:14]:
```suggestion
* **Ray Developers** write code that relies on Ray. They either run a single-node Ray Cluster locally or multi-node Clusters remotely on provided compute infrastructure.
```

Comment (ijrsvt) [reply doc/source/ray-security/index.md:14]: Yes!

File: doc/source/ray-security/index.md (hunk 2 above)
Comment (ijrsvt) [reply doc/source/ray-security/index.md:19]: No worries!

**Comment by angelinalg** [comment]
There are a few instances of passive voice that I didn't comment on. I'm assuming that is industry vernacular. But if it isn't, tI recommend you use the active voice consistently.

File: doc/source/ray-security/index.md (hunk 3 above)
Comment (ijrsvt) [reply doc/source/ray-security/index.md:22]: I may reword to "between core Ray components and additional Ray components". I'm not sure if these denote "Ray Core" per-se.

File: doc/source/ray-security/index.md [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+* **Platform providers** provide the compute environment on which **Developers** run Ray.
+* **Users** interact with the output of Ray-powered applications.
+
+## Best Practices
+**Security and isolation must be enforced outside of the Ray Cluster.** Ray expects to run in a safe network environment and to act upon trusted code. Developers and platform providers must maintain these invariants to ensure the safe operation of Ray Clusters:
+
+### Deploy Ray Clusters in a controlled network environment
+* Network traffic between Ray core components and Ray auxiliary components should always be in a controlled, isolated network. Access to auxiliary services should be gated with strict network controls and/or external authentication/authorization proxies.
```
Comment (ijrsvt) [inline doc/source/ray-security/index.md:22]:
```suggestion
* Network traffic between core Ray components and additional Ray components should always be in a controlled, isolated network. Access to additional services should be gated with strict network controls and/or external authentication/authorization proxies.
```

File: doc/source/ray-security/index.md [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+* gRPC communication can be encrypted with TLS, but it is not a replacement for network isolation.
+* Platform providers are responsible for ensuring that Ray runs in sufficiently controlled network environments and that developers can access features like Ray Dashboard in a secure manner.
+### Only execute trusted code within Ray
+* Ray faithfully executes code that is passed to it – Ray doesn’t differentiate between a tuning experiment, a rootkit install, or an S3 bucket inspection.
+* Ray developers are responsible for building their applications with this understanding in mind.
+### Enforce isolation outside of Ray with multiple Ray Clusters
+* If workloads require isolation from each other, use separate, isolated Ray Clusters. Ray can schedule multiple distinct Jobs in a single Cluster, but doesn't attempt to enforce isolation between them. Similarly, Ray doesn't implement access controls for developers interacting with a given cluster.
+* Ray developers hold the responsibility for determining which applications need to be separated and platform providers are responsible for providing this isolation.
```
Comment (ijrsvt) [inline doc/source/ray-security/index.md:30]:
```suggestion
* Ray developers are responsible for determining which applications need to be separated and platform providers are responsible for providing this isolation.
```

**Comment by ijrsvt** [comment]
Doc tests are passing & Pre-Merge has passed.

**Pull request merged**